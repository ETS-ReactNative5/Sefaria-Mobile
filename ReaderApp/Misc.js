'use strict';

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  Image,
  ActivityIndicator
} from 'react-native';

var Sefaria = require('./sefaria');

var styles = require('./Styles.js');


var TwoBox = React.createClass({
    propTypes: {
        content:  React.PropTypes.array.isRequired,
        language: React.PropTypes.oneOf(["hebrew","english"]),
    },
    render: function() {
        var content = this.props.content.map(function(item, i) {
            return (<View style={styles.twoBoxItem} key={i}>{item}</View>);
        });
        if (content.length % 2 !== 0) {
          content.push(<View style={styles.twoBoxItem} key={i+1}></View>);
        }
        var rows = [];
        var rowStyle = this.props.language == "hebrew" ? [styles.twoBoxRow, styles.rtlRow] : [styles.twoBoxRow];

        for (var i=0; i < content.length; i += 2) {
          var items = [content[i], content[i+1]];
          rows.push(<View style={rowStyle} key={i}>{items}</View>);
        }
        return (<View style={styles.twoBox}>{rows}</View>);
    }
});


var CategoryColorLine = React.createClass({
  render: function() {
    var style = {backgroundColor: Sefaria.palette.categoryColor(this.props.category)};
    return (<View style={[styles.categoryColorLine, style]}></View>);
  }
});


var LanguageToggleButton = React.createClass({
  propTypes: {
    theme:          React.PropTypes.object.isRequired,
    language:       React.PropTypes.string.isRequired,
    toggleLanguage: React.PropTypes.func.isRequired
  },
  render: function() {
    var content = this.props.language == "hebrew" ?
        (<Text style={[styles.languageToggleText, this.props.theme.languageToggleText, styles.en]}>A</Text>) : 
        (<Text style={[styles.languageToggleText, this.props.theme.languageToggleText, styles.he]}>◊ê</Text>);
    return (<TouchableOpacity style={[styles.languageToggle,this.props.theme.languageToggle]} onPress={this.props.toggleLanguage}>
              {content}
            </TouchableOpacity>);
  }
});


var SearchButton = React.createClass({
  render: function() {
    return (<TouchableOpacity style={[styles.headerButton]} onPress={this.props.onPress}>
              <Text style={[styles.searchButton, this.props.theme.searchButton]}>üîé</Text>
            </TouchableOpacity>);
  }
});


var MenuButton = React.createClass({
  render: function() {
    return (<TouchableOpacity style={[styles.headerButton]} onPress={this.props.onPress}>
              <Text style={[styles.menuButton, this.props.theme.menuButton]}>‚ò∞</Text>
            </TouchableOpacity>);
  }
});


var CloseButton = React.createClass({
  render: function() {
    return (<TouchableOpacity style={[styles.headerButton]} onPress={this.props.onPress}>
              <Text style={[styles.closeButton, this.props.theme.closeButton]}>√ó</Text>
            </TouchableOpacity>);
  }
});

var TripleDots = React.createClass({
  render: function() {
    return (<TouchableOpacity style={styles.tripleDotsContainer} onPress={this.props.onPress}>
              <Image style={styles.tripleDots}source={require('./img/dots.png')} />
            </TouchableOpacity>);
  }
});


var DisplaySettingsButton = React.createClass({
  render: function() {
    return (<TouchableOpacity style={[styles.headerButton]} onPress={this.props.onPress}>
              <Image source={require('./img/ayealeph.png')}
                     style={styles.displaySettingsButton}
                     resizeMode={Image.resizeMode.contain} />
            </TouchableOpacity>);
  }
});


var ToggleSet = React.createClass({
  propTypes: {
    theme:       React.PropTypes.object.isRequired,
    options:     React.PropTypes.array.isRequired, // array of object with `name`. `text`, `heText`, `onPress`
    contentLang: React.PropTypes.string.isRequired,
    active:      React.PropTypes.string.isRequired
  },
  render: function() {
    var showHebrew = this.props.contentLang == "hebrew";
    var options = this.props.options.map(function(option, i) {
      var style = [styles.navToggle, this.props.theme.navToggle].concat(this.props.active === option.name ? [styles.navToggleActive, this.props.theme.navToggleActive] : []);
      return (
        <TouchableOpacity onPress={option.onPress} key={i} >
          {showHebrew ?
            <Text style={[style, styles.he]}>{option.heText}</Text> :
            <Text style={[style, styles.en]}>{option.text}</Text> }
        </TouchableOpacity>
      );
    }.bind(this));

    var dividedOptions = [];
    for (var i = 0; i < options.length; i++) {
      dividedOptions.push(options[i])
      dividedOptions.push(<Text style={[styles.navTogglesDivider,this.props.theme.navTogglesDivider]} key={i+"d"}>|</Text>);
    }
    dividedOptions = dividedOptions.slice(0,-1);

    return (<View style={styles.navToggles}>
              {dividedOptions}
            </View>);
  }
});


var LoadingView = React.createClass({
    render: function() {
      return ( <View style={styles.loadingViewBox}>
                  <ActivityIndicator
                    animating={true}
                    style={styles.loadingView}
                    size="large" />
               </View> );
    }
});


module.exports.TwoBox = TwoBox;
module.exports.CategoryColorLine = CategoryColorLine;
module.exports.LanguageToggleButton = LanguageToggleButton;
module.exports.SearchButton = SearchButton;
module.exports.MenuButton = MenuButton;
module.exports.CloseButton = CloseButton;
module.exports.TripleDots = TripleDots;
module.exports.DisplaySettingsButton = DisplaySettingsButton;
module.exports.ToggleSet = ToggleSet;
module.exports.LoadingView = LoadingView;
