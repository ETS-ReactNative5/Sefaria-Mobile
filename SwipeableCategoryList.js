'use strict';

import PropTypes from 'prop-types';

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
  ScrollView,
  Alert,
  SwipeableFlatList,
  FlatList,
  Image,
  RefreshControl,
} from 'react-native';
import {
  CategoryColorLine,
  CategorySideColorLink,
  DirectedButton,
  TwoBox,
  LanguageToggleButton,
  AnimatedRow,
  SText,
} from './Misc.js';
import { STATE_ACTIONS } from './StateManager';
import styles from './Styles';
import strings from './LocalizedStrings';


class SwipeableCategoryList extends React.Component {
  static propTypes = {
    close:              PropTypes.func.isRequired,
    theme:              PropTypes.object.isRequired,
    themeStr:           PropTypes.string.isRequired,
    openRef:            PropTypes.func.isRequired,
    defaultTextLanguage:PropTypes.oneOf(["english","bilingual", "hebrew"]).isRequired,
    interfaceLanguage:  PropTypes.oneOf(["english","hebrew"]),
    onRemove:           PropTypes.func,
    title:              PropTypes.string.isRequired,
    loadData:           PropTypes.func.isRequired,
    icon:               PropTypes.number.isRequired,
    menuOpen:           PropTypes.oneOf(["saved", "history"]),
    openLogin:          PropTypes.func.isRequired,
    isLoggedIn:         PropTypes.bool.isRequired,
    hasDismissedSyncModal: PropTypes.bool.isRequired,
    dispatch:           PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = {
      data: [],
      refreshing: true,
    };
    this._rowRefs = {};
  }

  componentDidMount() { this.loadData(); }

  loadData = async () => {
    let data = await this.props.loadData();
    // reduce consecutive history items with the same ref
    // filter items that are marked `secondary`
    data = data.reduce(
      (accum, curr, index) => (
        (!accum.length || curr.ref !== accum[accum.length-1].ref) ?
          accum.concat([curr]) :
          accum
      ), []
    ).filter(h => !h.secondary);
    this.setState({ data, refreshing: false });
  }

  removeItem = (item) => {
    const ref = this._rowRefs[item.ref];
    if (ref) {
      ref.remove();
    }
  }

  onRefresh = () => { this.setState({ refreshing: true }, this.loadData); }

  _getRowRef = (ref, item) => {
    this._rowRefs[item.ref] = ref;
  }

  renderDeleteButton = ({ item }) => (
    <View style={[{flex:1}, this.props.theme.secondaryBackground]}>
      <TouchableOpacity onPress={() => { this.removeItem(item); }} style={{alignSelf: 'flex-end', justifyContent: 'center', flex:1, width:90}}>
        <Text style={[{textAlign: 'center'}, this.props.theme.contrastText]}>
          {strings.remove}
        </Text>
      </TouchableOpacity>
    </View>
  );

  renderRow = ({ item }) => (
      <AnimatedRow
        ref={ref => { this._getRowRef(ref, item); }}
        animationDuration={250}
        onRemove={() => { this.props.onRemove(item); }}
        style={{flex: 1, justifyContent: "center", alignItems: "center"}}
      >
        <CategorySideColorLink
          category={Sefaria.categoryForTitle(item.book)}
          enText={item.ref}
          heText={item.he_ref}
          language={this.props.defaultTextLanguage}
          onPress={this.props.openRef.bind(null, item.ref, null, item.versions)}
        />
      </AnimatedRow>
  );

  _keyExtractor = (item, index) => (
    `${item.ref}|${item.time_stamp}`
  );

  render() {
    const FlatListClass = this.props.menuOpen === "history" ? FlatList : SwipeableFlatList;  // disable swiping on history
    const isWhite = this.props.themeStr === "white";
    const isHeb = this.props.interfaceLanguage === "hebrew";
    return (
      <View style={[styles.menu, this.props.theme.menu]}>
        <CategoryColorLine category={"Other"} />
        <View style={[styles.header, this.props.theme.header]}>
          <DirectedButton
            onPress={this.props.close}
            imageStyle={[styles.menuButton, styles.directedButton]}
            direction="back"
            language="english"/>
          <View style={{flex:1, flexDirection: "row", justifyContent: "center", alignItems: "center"}}>
            <Image source={this.props.icon}
              style={[styles.menuButton, isHeb ? styles.headerIconWithTextHe : styles.headerIconWithTextEn]}
              resizeMode={'contain'}
            />
            <SText
              lang={this.props.interfaceLanguage}
              style={[styles.textTocHeaderTitle, {flex:0},styles.noPadding, this.props.theme.text]}>
              {this.props.title.toUpperCase()}
            </SText>
          </View>
          <LanguageToggleButton />
        </View>
        {this.props.isLoggedIn || this.props.hasDismissedSyncModal ? null :
          <TouchableOpacity style={{
              backgroundColor: "#18345D",
              paddingVertical: 20,
              paddingHorizontal: 15,
              flexDirection: 'row',
              alignItems: 'center',
              justifyContent: 'space-between',
            }}
            onPress={this.props.openLogin}
          >
            <Text style={[ styles.systemButtonText, styles.systemButtonTextBlue, styles.enInt]}>
              { `${strings.wantToSync} ` }
              <Text style={[{ textDecorationLine: 'underline'}]}>{ strings.logIn }</Text>
            </Text>

            <TouchableOpacity onPress={() => {
                this.props.dispatch({
                  type: STATE_ACTIONS.setHasDismissedSyncModal,
                  value: true,
                });
              }}>
              <Image
                source={require('./img/close-light.png')}
                resizeMode={'contain'}
                style={{width: 14, height: 14}}
              />
            </TouchableOpacity>
          </TouchableOpacity>
        }
        <FlatListClass
          data={this.state.data}
          renderItem={this.renderRow}
          keyExtractor={this._keyExtractor}
          bounceFirstRowOnMount={!Sefaria._hasSwipeDeleted}
          maxSwipeDistance={90}
          renderQuickActions={this.renderDeleteButton}
          language={this.props.defaultTextLanguage}
          refreshControl={
            <RefreshControl
              refreshing={this.state.refreshing}
              onRefresh={this.onRefresh}
              tintColor="#CCCCCC"
              style={{ backgroundColor: 'transparent' }} />
          }
        />
      </View>
    );
  }
}

export default SwipeableCategoryList;
