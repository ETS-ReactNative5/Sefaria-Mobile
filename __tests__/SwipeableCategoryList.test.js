import React from 'react';
import renderer from 'react-test-renderer';
import SwipeableCategoryList from '../SwipeableCategoryList';
import { AnimatedRow } from '../Misc';

describe('SwipeableCategoryList', () => {
  test('basic render', async () => {
    const tree = renderer.create(
      <SwipeableCategoryList
        close={() => {}}
        theme={{}}
        themeStr={'white'}
        toggleLanguage={() => {}}
        openRef={() => {}}
        language={'english'}
        interfaceLang={'english'}
        onRemove={() => {}}
        title={'History'}
        loadData={() => Promise.resolve([])}
        menuOpen={'history'}
        icon={require('./img/clock.png')}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('render with data', async () => {
    Sefaria.categoryForTitle = jest.fn(() => "Tanakh");
    const inst = renderer.create(
      <SwipeableCategoryList
        close={() => {}}
        theme={{}}
        themeStr={'white'}
        toggleLanguage={() => {}}
        openRef={() => {}}
        language={'english'}
        interfaceLang={'english'}
        onRemove={() => {}}
        title={'History'}
        loadData={() => Promise.resolve([
          {  // normal case with versions
            ref: "Genesis 1:1",
            he_ref: "בראשית א:א",
            versions: { en: "enVersion", he: "heVersion" },
            book: "Genesis"
          },
          {  // no versions
            ref: "Genesis 1:2",
            he_ref: "בראשית א:ב",
            versions: {},
            book: "Genesis"
          },
          {  // section ref
            ref: "Genesis 1",
            he_ref: "בראשית א",
            versions: {},
            book: "Tanakh"
          },
          {  // duplicate
            ref: "Genesis 1",
            he_ref: "בראשית א",
            versions: {},
            book: "Tanakh"
          },
          {  // secondary
            ref: "Genesis 1",
            he_ref: "בראשית א",
            versions: {},
            book: "Tanakh",
            secondary: true,
          }
        ])}
        menuOpen={'history'}
        icon={require('./img/clock.png')}
      />
    );
    await setTimeout(()=>{}, 10);  // wait for SwipeableFlatList to fully render
    const yo = inst.root.findAllByType(AnimatedRow);
    expect(yo.length).toBe(3);
    expect(inst.toJSON()).toMatchSnapshot();
  });
});
