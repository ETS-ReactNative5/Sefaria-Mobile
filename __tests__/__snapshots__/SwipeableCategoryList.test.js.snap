// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwipeableCategoryList basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "height": 8,
        },
        Object {
          "backgroundColor": "#073570",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 50,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 10,
          "right": 10,
          "top": 20,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row-reverse",
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 16,
              "width": 16,
            },
            Object {
              "marginLeft": 12,
              "marginRight": 12,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 16,
              "width": 16,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 8,
            },
          ]
        }
      />
      <Text
        lang="english"
        style={
          Array [
            Object {
              "flex": 1,
              "letterSpacing": 1,
              "paddingLeft": 10,
              "textAlign": "center",
            },
            Object {
              "flex": 0,
            },
            Object {
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
            undefined,
            Object {
              "lineHeight": 24,
            },
          ]
        }
      >
        HISTORY
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "fontSize": 19,
              "marginRight": 2,
              "marginTop": 4,
            },
            undefined,
            Object {
              "fontFamily": "Taamey Frank Taamim Fix",
              "textAlign": "right",
            },
          ]
        }
      >
        א
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flex": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Object {
          "backgroundColor": "transparent",
          "height": 80,
        }
      }
    />
  </View>
</View>
`;

exports[`SwipeableCategoryList render with data 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "height": 8,
        },
        Object {
          "backgroundColor": "#073570",
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "height": 50,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      hitSlop={
        Object {
          "bottom": 20,
          "left": 10,
          "right": 10,
          "top": 20,
        }
      }
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row-reverse",
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 16,
              "width": 16,
            },
            Object {
              "marginLeft": 12,
              "marginRight": 12,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Array [
            Object {
              "height": 16,
              "width": 16,
            },
            Object {
              "marginLeft": 0,
              "marginRight": 8,
            },
          ]
        }
      />
      <Text
        lang="english"
        style={
          Array [
            Object {
              "flex": 1,
              "letterSpacing": 1,
              "paddingLeft": 10,
              "textAlign": "center",
            },
            Object {
              "flex": 0,
            },
            Object {
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 0,
            },
            undefined,
            Object {
              "lineHeight": 24,
            },
          ]
        }
      >
        HISTORY
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 30,
          "justifyContent": "center",
          "marginRight": 20,
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "black",
              "fontSize": 19,
              "marginRight": 2,
              "marginTop": 4,
            },
            undefined,
            Object {
              "fontFamily": "Taamey Frank Taamim Fix",
              "textAlign": "right",
            },
          ]
        }
      >
        א
      </Text>
    </View>
  </View>
  <RCTScrollView
    bounceFirstRowOnMount={true}
    data={
      Array [
        Object {
          "book": "Genesis",
          "he_ref": "בראשית א:א",
          "ref": "Genesis 1:1",
          "versions": Object {
            "en": "enVersion",
            "he": "heVersion",
          },
        },
        Object {
          "book": "Genesis",
          "he_ref": "בראשית א:ב",
          "ref": "Genesis 1:2",
          "versions": Object {},
        },
        Object {
          "book": "Tanakh",
          "he_ref": "בראשית א",
          "ref": "Genesis 1",
          "versions": Object {},
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    language="english"
    maxSwipeDistance={90}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    renderQuickActions={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "height": 50,
              "left": 0,
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderLeftColor": "#004e5f",
                    "borderLeftWidth": 6,
                    "width": 6,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "flex": 1,
                      "paddingBottom": 10,
                      "paddingHorizontal": 48,
                      "paddingTop": 14,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="middle"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontFamily": "Amiri",
                        "fontSize": 19,
                        "lineHeight": 26,
                        "marginTop": 5,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Genesis 1:1
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "height": 50,
              "left": 0,
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderLeftColor": "#004e5f",
                    "borderLeftWidth": 6,
                    "width": 6,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "flex": 1,
                      "paddingBottom": 10,
                      "paddingHorizontal": 48,
                      "paddingTop": 14,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="middle"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontFamily": "Amiri",
                        "fontSize": 19,
                        "lineHeight": 26,
                        "marginTop": 5,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Genesis 1:2
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "height": 50,
              "left": 0,
              "opacity": 1,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "borderLeftColor": "#004e5f",
                    "borderLeftWidth": 6,
                    "width": 6,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 1,
                      "flex": 1,
                      "paddingBottom": 10,
                      "paddingHorizontal": 48,
                      "paddingTop": 14,
                    },
                    undefined,
                    undefined,
                  ]
                }
              >
                <Text
                  ellipsizeMode="middle"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "fontFamily": "Amiri",
                        "fontSize": 19,
                        "lineHeight": 26,
                        "marginTop": 5,
                        "textAlign": "left",
                      },
                      undefined,
                    ]
                  }
                >
                  Genesis 1
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
